### PR TITLE
icount mini segment

### LINKED ISSUE #3370: new module: ICOUNT-MINI/SEGMENT
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by CharlotteAnne**
## PR checklist

Closes #3370  

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/icountmini/segment/main.nf [hunk 1]
```diff
@@ -0,0 +1,40 @@
+process ICOUNTMINI_SEGMENT {
+    tag "$gtf"
+    label "process_single"
+
+    conda "bioconda::icount-mini=2.0.3"
+
```
Comment (chris-cheshire) [inline modules/nf-core/icountmini/segment/main.nf:6]: remove line

File: modules/nf-core/icountmini/segment/main.nf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/icount-mini:2.0.3--pyh5e36f6f_0' :
+        'quay.io/biocontainers/icount-mini:2.0.3--pyh5e36f6f_0' }"
+
+    input:
+    tuple val(meta), path(gtf)
+    path fai
+
+    output:
+    tuple val(meta), path("*_seg.gtf")                  ,  emit: gtf
```
Comment (chris-cheshire) [inline modules/nf-core/icountmini/segment/main.nf:16]: Sort out whitespace

File: tests/config/test_data.config [hunk 3]
```diff
@@ -184,6 +184,7 @@ params {
                 genome_21_sdf                                  = "${params.test_data_base}/data/genomics/homo_sapiens/genome/chr21/sequence/genome_sdf.tar.gz"
                 genome_21_fasta                                = "${params.test_data_base}/data/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta"
                 genome_21_fasta_fai                            = "${params.test_data_base}/data/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai"
+                genome_21_gtf                                  = "${params.test_data_base}/data/genomics/homo_sapiens/genome/chr21/sequence/chr21_gencode.gtf"
```
Comment (chris-cheshire) [inline tests/config/test_data.config:187]: Same comment as the clippy PR

**Review by chris-cheshire — CHANGES_REQUESTED** [review]

File: modules/nf-core/icountmini/segment/main.nf (hunk 2 above)
Comment (CharlotteAnne) [reply modules/nf-core/icountmini/segment/main.nf:16]: again, not seeing the issue, sorry?

File: tests/config/test_data.config (hunk 3 above)
Comment (CharlotteAnne) [reply tests/config/test_data.config]: updated

**Review by chris-cheshire — APPROVED** [review]

**Pull request merged**